### DISCUSSION TITLE
Vercel and firebase private key issue

### DISCUSSION THREAD

**Discussion by jamiegood** [gladly-team/next-firebase-auth#417] [Q&A] [answered]
HI all, 

wanted to ask
I am currently using Vercel for my nextjs app. Unfortunately Vercel as a max 4kb size for env vars. 
This means that I can't fit the FIREBASE_PRIVATE_KEY into a env var.
One approach: I encrypt the Firebase Private Key with AES. And store the key secret/iv in env vars.(smaller)
Then at run time decrypt to get the Firebase Private Key and configure firebaseAdmin SDK with that. 

So basically I'd need to set the config for firebaseAdminInitConfig server side as part of a /api/route or similar. 
But at the moment I think all the config client and server side are set together. 
Is it possible to set them separately to allow me to set the client side via react and server side (firebaseAdminInitConfig) somewhere else?

To avoid the following message: "Error: Invalid next-firebase-auth options: The "firebaseAdminInitConfig" private key setting should not be available on the client side."

Feedback appreciated!
Many thanks 

This is the config: 
`    firebaseAdminInitConfig: {
      credential: {
        projectId: 'my-example-app-id',
        clientEmail: 'example-abc123@my-example-app.iam.gserviceaccount.com',
        // The private key must not be accessible on the client side.
        privateKey: process.env.FIREBASE_PRIVATE_KEY,
      },
      databaseURL: 'https://my-example-app.firebaseio.com',`

**Comment by kmjennison** [comment, marked as answer]
When validating whether the private key is set or not set, this library only cares about the following:
* on the server side, the value **must** be defined
* on the client side, the value **cannot** be defined

Though the config is initialized in the same code for the server and client, it can (and should) contain different values.

It seems like the difficult part in your case is fetching the private key value prior to loading the app. You might approach this with middleware or a higher-order function that gets the private key server-side and passes it to the composed function that initializes `next-firebase-auth`.

**Reply by jamiegood** [reply]
Thanks for the reply. Will look into this approach.
